### PR TITLE
Add integration with Sonar

### PR DESCRIPTION
**Pull request by chudilka1**

### CODE REVIEW

**Comment by chudilka1** [comment]
Currently, the removed lint workflow is always in `Waiting for status to be reported`. It would be necessary to go update the configuration of branch protection rules as follows [here](https://github.com/orgs/community/discussions/26698#discussioncomment-3252954)

**Review by krehermann — COMMENTED** [review]
are there any significant difference between this integration and others that you have already done?


how do you test that it's working as expected?

**Review by patrickhuie19 — COMMENTED** [review]
IMO, a fix for the lint error pertaining to the potential slowloris attack should be addressed and merged before this PR. I can handle

update: https://github.com/smartcontractkit/wsrpc/pull/39

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,17 +1,73 @@
...
+          # Optional: if set to true then the action will use pre-installed Go.
+          # skip-go-installation: true
+
+          # Optional: if set to true then the action don't cache or restore ~/go/pkg.
+          # skip-pkg-cache: true
+
+          # Optional: if set to true then the action don't cache or restore ~/.cache/go-build.
+          # skip-build-cache: true
+
+      - name: Verify lint report artifact
```
Comment (jmank88) [inline .github/workflows/ci.yml:47]: Why 'Verify'? This is about printing the results for humans, if the file exists.
```suggestion
      - name: Print lint report artifact
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -21,5 +77,48 @@ jobs:
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go
+      
       - name: Run Test Suite
-        run: go test -p=1 -race ./...
+        run: go test -p=1 -race ./... -coverpkg=./... -coverprofile=coverage.txt $1 | tee $OUTPUT_FILE
```
Comment (jmank88) [inline .github/workflows/ci.yml:82]: Perhaps out of scope, but why is this using `-p=1` to test one package at a time?
```suggestion
        run: go test -race ./... -coverpkg=./... -coverprofile=coverage.txt $1 | tee $OUTPUT_FILE
```

Comment (chudilka1) [reply .github/workflows/ci.yml]: Thanks, removed `-p` flag

**Pull request merged**